lib/ex_unit/test/ex_unit/assertions_test.exs

```diff
@@ -336,6 +336,32 @@ defmodule ExUnit.AssertionsTest do
       "Expected exception SyntaxError but got FunctionClauseError (no function clause matching in :lists.flatten/1)" = error.message
   end
 
+  test "assert raise comparing messages (for equality)" do
+    assert_raise RuntimeError, "foo", fn ->
+      raise RuntimeError, "bar"
+    end
+  rescue
+    error in [ExUnit.AssertionError] ->
+      "Wrong message for RuntimeError" <>
+      "\nExpected:" <>
+      "\n  \"foo\"" <>
+      "\nGot:" <>
+      "\n  \"bar\"" = error.message
+  end
+
+  test "assert raise comparing messages (with a regex)" do
+    assert_raise RuntimeError, ~r/ba[zk]/, fn ->
+      raise RuntimeError, "bar"
+    end
+  rescue
+    error in [ExUnit.AssertionError] ->
+      "Wrong message for RuntimeError" <>
+      "\nExpected:" <>
+      "\n  ~r/ba[zk]/" <>
+      "\nGot:" <>
+      "\n  \"bar\"" = error.message
```

Comment:
Just to give a little bit more of context, 
besides the [lines that you took as reference](https://github.com/elixir-lang/elixir/blob/master/lib/ex_unit/test/ex_unit/assertions_test.exs#L157-L161), there is only one line in another test that uses that convention, and that's it for the tests.
https://gist.github.com/eksperimental/68403926802cb6a967da#file-leading_n-log-L2-L4

the rest of the cases are 3 .ex files using it, and in one of them the usage is proper, i think the other two it can be corrected.